src/Microsoft.OData.Core/ContainerBuilderExtensions.cs

```diff
@@ -136,7 +136,7 @@ public static IContainerBuilder AddDefaultODataServices(this IContainerBuilder b
             builder.AddService<IJsonWriterFactoryAsync>(ServiceLifetime.Singleton, sp => new DefaultJsonWriterFactory());
             builder.AddService(ServiceLifetime.Singleton, sp => ODataMediaTypeResolver.GetMediaTypeResolver(null));
             builder.AddService<ODataMessageInfo>(ServiceLifetime.Scoped);
-            builder.AddServicePrototype(new ODataMessageReaderSettings());
+            builder.AddServicePrototype(new ODataMessageReaderSettings(odataVersion));
             builder.AddService(ServiceLifetime.Scoped, sp => sp.GetServicePrototype<ODataMessageReaderSettings>().Clone());
             builder.AddServicePrototype(new ODataMessageWriterSettings());

```

Comment:
@xuzhg Can we do it this way for `ODataMessageWriterSettings`?
```suggestion
            builder.AddServicePrototype(new ODataMessageWriterSettings { Version = odataVersion });
```